src/main/java/org/commcare/cases/entity/NodeEntityFactory.java

```diff
@@ -192,4 +194,34 @@ public final boolean isEntitySetReady() {
             return isEntitySetReadyInternal();
         }
     }
+
+    /**
+     * Caches the provided entities. Default implementation throws RuntimeException.
+     * Subclasses should override this method if they support caching.
+     *
+     * @param entities List of entities to cache
+     * @throws RuntimeException if caching is not supported
+     */
+    public void cacheEntities(List<Entity<TreeReference>> entities) {
+        throw new RuntimeException("Method not supported for normal Node Entity Factory");
+    }
+
+    /**
+     * Sets the progress listener for entity loading operations.
+     *
+     * @param progressListener The progress listener to use
+     */
+    public void setEntityProgressListener(EntityLoadingProgressListener progressListener) {
+        this.progressListener = progressListener;

```

Comment:
If there is an existing listener set, I think this method should close it out before changing the current one. Alternatively this method should potentially error out, or otherwise treat the listener as immutable once set.

Reply:
https://github.com/dimagi/commcare-core/pull/1454/commits/5a1422e017dec6897375e3c0f074982593f0d7dc